Apps/W1/HybridGP/app/src/Migration/History/GPPopulateHistTables.Codeunit.al

```diff
@@ -390,13 +390,52 @@ codeunit 40900 "GP Populate Hist. Tables"
             HistSalesTrxLine."Location Code" := GPSOPTrxAmountsHist.LOCNCODE;
             HistSalesTrxLine."Ship-to Name" := CopyStr(GPSOPTrxAmountsHist.ShipToName, 1, MaxStrLen(HistSalesTrxLine."Ship-to Name"));
 
-            if not HistSalesTrxLine.Insert() then
+            if HistSalesTrxLine.Insert() then
+                PopulateSalesLineTrxSerialLotDetail(HistSalesTrxLine, GPSOPTrxAmountsHist)
+            else
                 ReportLastError(Database::GPSOPTrxAmountsHist, GPSOPTrxAmountsHist.DEX_ROW_ID, "Hist. Migration Step Type"::"GP Receivables Trx.", GPSOPTrxHist.SOPNUMBE);
 
             AfterProcessedNextChildRecord();
         until GPSOPTrxAmountsHist.Next() = 0;
     end;
 
+    local procedure PopulateSalesLineTrxSerialLotDetail(var HistSalesTrxLine: Record "Hist. Sales Trx. Line"; var GPSOPTrxAmountsHist: Record GPSOPTrxAmountsHist)
+    var
+        GPSOPSerialLotWorkHist: Record GPSOPSerialLotWorkHist;
+        HistRecvTrxSerialLot: Record "Hist. Recv. Trx. SerialLot";
+    begin
+        GPSOPSerialLotWorkHist.SetLoadFields(SOPTYPE, SOPNUMBE, LNITMSEQ, POSTED, SERLTNUM, SERLTQTY, DATERECD, DTSEQNUM, UNITCOST, CMPNTSEQ, SLTSQNUM, OVRSERLT, BIN, MFGDATE, EXPNDATE, TRXSORCE);
```

Comment:
In general SetLoadFields is not needed for these tables. It can also mess up the extensibility story. See if you really need to limit the fields that are set. We usually set it to get a performance improvement if there are too many extensions in play.

Reply:
The snapshot deals with millions of records and SetLoadFields is used throughout for performance reasons